dot/core/service.go

```diff
@@ -574,145 +592,20 @@ func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
-			if err != nil {
-				return err
-			}
-
-			// get block header; if exists, return
-			existingHeader, err := s.blockState.GetHeader(bd.Hash)
-			if err != nil && existingHeader == nil {
-				err = s.blockState.SetHeader(header)
-				if err != nil {
-					return err
-				}
-
-				log.Info("[core] saved block header", "hash", header.Hash(), "number", header.Number)
-
-				// TODO: handle consensus digest, if first in epoch
-				// err = s.handleConsensusDigest(header)
-				// if err != nil {
-				// 	return err
-				// }
-			}
-		}
-
-		if bd.Header.Exists() && bd.Body.Exists {
-			header, err := types.NewHeaderFromOptional(bd.Header)
-			if err != nil {
-				return err
-			}
-
-			body, err := types.NewBodyFromOptional(bd.Body)
-			if err != nil {
-				return err
-			}
-
-			block := &types.Block{
-				Header: header,
-				Body:   body,
-			}
-
-			// TODO: why doesn't execute block work with block we built?
-
-			// blockWithoutDigests := block
-			// blockWithoutDigests.Header.Digest = [][]byte{{}}
-
-			// enc, err := block.Encode()
-			// if err != nil {
-			// 	return err
-			// }
-
-			// err = s.executeBlock(enc)
-			// if err != nil {
-			// 	log.Error("[core] failed to validate block", "err", err)
-			// 	return err
-			// }
-
-			if header.Number.Cmp(bestNum) == 1 {
-				err = s.blockState.AddBlock(block)
-				if err != nil {
-					log.Error("[core] Failed to add block to state", "error", err, "hash", header.Hash(), "parentHash", header.ParentHash)
-					return err
-				}
-
-				log.Info("[core] imported block", "number", header.Number, "hash", header.Hash())
-
-				err = s.checkForRuntimeChanges()
-				if err != nil {
-					return err
-				}
-			}
-		}
-
-		err := s.compareAndSetBlockData(bd)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-func (s *Service) compareAndSetBlockData(bd *types.BlockData) error {
-	if s.blockState == nil {
-		return fmt.Errorf("no blockState")
-	}
-
-	existingData, err := s.blockState.GetBlockData(bd.Hash)
-	if err != nil {
-		// no block data exists, ok
-		return s.blockState.SetBlockData(bd)
-	}
-
-	if existingData == nil {
-		return s.blockState.SetBlockData(bd)
-	}
-
-	if existingData.Header == nil || (!existingData.Header.Exists() && bd.Header.Exists()) {
-		existingData.Header = bd.Header
-	}
-
-	if existingData.Body == nil || (!existingData.Body.Exists && bd.Body.Exists) {
-		existingData.Body = bd.Body
-	}
-
-	if existingData.Receipt == nil || (!existingData.Receipt.Exists() && bd.Receipt.Exists()) {
-		existingData.Receipt = bd.Receipt
-	}
-
-	if existingData.MessageQueue == nil || (!existingData.MessageQueue.Exists() && bd.MessageQueue.Exists()) {
-		existingData.MessageQueue = bd.MessageQueue
-	}
-
-	if existingData.Justification == nil || (!existingData.Justification.Exists() && bd.Justification.Exists()) {
-		existingData.Justification = bd.Justification
-	}
+	log.Debug("[core] received BlockResponseMessage")
+	s.respChan <- msg.(*network.BlockResponseMessage)
```

Comment:
if only appended to syncer, we can use `s.syncer.respChan` herre, which might help make it more clear the `respChan` is specific to the `syncer` submodule

Reply:
that wouldn't work since syncer.respChan is read only and service.respChan is write only

Reply:
right, maybe `syncer.respIn` and `syncer.respOut`? they are both specific to the syncer, correct?

Reply:
hmm like rename the Service one to `respOut`?

Reply:
Yea, I think the naming is confusing me a bit.

Can we use the syncer's write only channel rather than appending a new channel to the service called `respChan` (using the syncer more like a submodule that provides both read only and write only channels to the services which the `syncer` is appended to).

We already have `syncer.msgIn` (which receives messages within syncer from core service and network service) and `syncer.msgOut` (which sends messages within syncer to network service) and the syncer is being used by both the core service and the network service like a submodule.

It would be nice if we only needed to append `syncer` to each service, which in itself (like a submodule) provided all in and out channels specific to the syncer. Is that possible?

Reply:
thinking out loud a bit so might not make sense...

Reply:
okay I see what you mean.. I could change it to a read and write channel in syncer, so that core could write to it then syncer could read from it.. or I could separate them within syncer. I just didn't do that since having a write only channel within syncer sort of suggests that syncer writes to it. but I can change it

Reply:
I think it would make more sense to have the separation. At the least, we should probably add comments wherever we use channels explaining the ins and outs of each.